pkg/web/webtest/webtest.go

```diff
@@ -0,0 +1,120 @@
+package webtest
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+var requests = map[string]*models.ReqContext{}
+
+type Server struct {
+	t             *testing.T
```

Comment:
```suggestion
	t             testing.TB
```

here and in the NewServer declaration should make it support benchmarks as well